test/support/pulp/repository_support.rb

```diff
@@ -30,8 +30,8 @@ def self.create_and_sync_repo(repo)
     def self.create_repo(repo)
       FactoryBot.create(:smart_proxy, :default_smart_proxy) unless ::SmartProxy.pulp_master
 
-      repo.relative_path = repo.puppet? ? PULP_TMP_DIR : 'test_path'
-      repo.root.url = repo.puppet? ? @puppet_repo_url : @repo_url
+      repo.relative_path = (repo.puppet? ? PULP_TMP_DIR : 'test_path') unless repo.file?
+      repo.root.url = (repo.puppet? ? @puppet_repo_url : @repo_url) unless repo.file?
       repo.root.download_policy = :immediate if repo.yum?
```

Comment:
We specify relative path and sync url in file fixtures so we do not need to overwrite these for files. I am not sure if we have tests saving other types of repos but doesn't seem to affect them.